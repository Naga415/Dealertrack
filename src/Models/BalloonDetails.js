'use strict';

export default class BalloonDetails {
  constructor() {
    this.msrp = null;
    this.sellingPrice = null;
    this.addOns = null;
    this.fees = null;
    this.taxes = null;
    this.rebate = null;
    this.rate = null;
    this.term = null;
    this.firstPayment = null;
    this.downpayments = [];
    this.balloon = null;
    this.lenderFee = null;
  }
}
