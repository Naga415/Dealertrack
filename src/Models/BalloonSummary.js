'use strict';
export default class BalloonSummary {
  constructor() {
    this.payment = null;
    this.amountFinananced = null;
    this.holdDeposit = null;
    this.purchasePriceWithAdds = null;
    this.discountFromSellingPrice = null;
    this.netTrade = null;
    this.balloonAmount = null;
    this.interestRateTotal = null;
    this.totalPayments = null;
    this.subtotal = null;
  }
}
