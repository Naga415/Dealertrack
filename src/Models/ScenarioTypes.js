export const FINANCE = "Finance";
export const LEASE = "Lease";
export const CASH = "Cash";
export const BALLOON = "Balloon";
