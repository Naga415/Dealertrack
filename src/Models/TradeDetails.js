'use strict';
export default class TradeDetails {
  constructor() {
    this.allowance = null;
    this.payoff = null;
    this.acv = null;
  }
}
