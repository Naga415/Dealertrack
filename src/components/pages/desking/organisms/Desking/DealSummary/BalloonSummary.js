'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import Summary from '../../../molecules/Summary/Summary';
import BalloonSummaryModel from '../../../../../../models/BalloonSummary';

// eslint-disable-next-line react/prefer-stateless-function
class BalloonSummary extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }

  render() {
    return (
      <Summary title="Balloon Deal" htmlId="BalloonSummary">
        <Summary.Item label="Hold Deposit" value={this.props.financialSummary.holdDeposit}/>
        <Summary.Item label="Purchase Price w/ Adds" value={this.props.financialSummary.purchasePriceWithAdds}/>
        <Summary.Item label="Discount from Selling Price" value={this.props.financialSummary.discountFromSellingPrice}/>
        <Summary.Item label="Net Trade" value={this.props.financialSummary.netTrade}/>
        <Summary.Item label="Balloon Amount" value={this.props.financialSummary.balloonAmount}/>
        <Summary.Item label="Interest Rate Total" value={this.props.financialSummary.interestRateTotal}/>
        <Summary.Item label="Total of Payments" value={this.props.financialSummary.totalPayments}/>
      </Summary>
    );
  }
}

BalloonSummary.propTypes = {
  financialSummary: PropTypes.instanceOf(BalloonSummaryModel)
};

export default BalloonSummary;
