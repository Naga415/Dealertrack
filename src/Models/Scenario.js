'use strict';
import TradeDetails from './TradeDetails';

export default class Scenario {
  constructor() {
    this.id = null;
    this.type = null;
    this.summary = null;
    this.details = null;
    this.trade = new TradeDetails();
    this.frontProfitBreakdown = [];
    this.backProfitBreakdown = [];
  }
}
