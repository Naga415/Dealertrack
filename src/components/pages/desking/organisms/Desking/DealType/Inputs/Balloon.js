"use strict";
import React from "react";
import PropTypes from "prop-types";
import Stacker from "../../../../molecules/Stacker/Stacker";
import Input from "../../../../molecules/Input/Input";
import ReadOnly from "../../../../molecules/ReadOnly/ReadOnly";
import Scenario from "../../../../../../../models/Scenario";
import { Taxes } from "@coxautokc/taxes";
import { Fees } from "@coxautokc/fees";
import { Rebates } from '@coxautokc/rebates';

class Balloon extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
    this.state = {
      showFeesModal: false,
      showRebatesModal: false,
      showTaxesModal: false
    };
  }

  onDetailsChange = e => {
    const { name, value } = e.target;
    this.props.actions.deal.updateScenarioDetails(this.props.scenario.id, {
      name,
      value
    });
  };

  onTradeChange = e => {
    const { name, value } = e.target;
    this.props.actions.deal.updateScenarioTrade(this.props.scenario.id, {
      name,
      value
    });
  };

  onClick = () => {};

  // Begin: Integration for Fees, Rebates, Taxes

  openCloseModal = modalName => {
    const tmp = {};
    this.setState(function(prevState) {
      tmp[modalName] = !prevState[modalName];
      return Object.assign({}, prevState, tmp);
    });
  };

  onDialogChange = (name, value) => {
    this.props.actions.deal.updateScenarioDetails(this.props.scenario.id, {
      name,
      value
    });
  };

  // End: Integration for Fees, Rebates, Taxes

  render() {
    return (
      <div id={this.htmlId} className="deal-type-inputs">
        <Stacker>
          <Stacker.Stackable className="deal-type-inputs__stackable">
            <Input
              id="MSRP"
              name="msrp"
              value={this.props.scenario.details.msrp}
              label="MSRP"
              onChange={this.onDetailsChange}
              onBlur={this.props.save}
              tabIndex={1}
            />
          </Stacker.Stackable>
          <Stacker.Stackable className="deal-type-inputs__stackable">
            <Input
              id="Balloon"
              name="Balloon"
              value={this.props.scenario.details.balloon}
              label="Balloon"
              options={[{ label: "Percent" }, { label: "Dollar Amount" }]}
              onChange={this.onDetailsChange}
              onBlur={this.props.save}
              tabIndex={9}
            />
          </Stacker.Stackable>
          <Stacker.Stackable className="deal-type-inputs__stackable">
            <Input
              id="SellingPrice"
              name="sellingPrice"
              value={this.props.scenario.details.sellingPrice}
              label="Selling Price"
              onChange={this.onDetailsChange}
              onBlur={this.props.save}
              tabIndex={2}
            />
          </Stacker.Stackable>
          <Stacker.Stackable className="deal-type-inputs__stackable">
            <Input
              id="Downpayment.0"
              name="downpayments.0"
              value={this.props.scenario.details.downpayments[0]}
              label="Down Payment"
              options={[{ label: "Percent" }, { label: "Dollar Amount" }]}
              onChange={this.onDetailsChange}
              onBlur={this.props.save}
              tabIndex={10}
            />
          </Stacker.Stackable>
          <Stacker.Stackable className="deal-type-inputs__stackable">
            <Input
              id="AddOns"
              name="addOns"
              value={this.props.scenario.details.addOns}
              label="Add-Ons"
              onClick={this.onClick}
              onChange={this.onDetailsChange}
              onBlur={this.props.save}
              tabIndex={3}
            />
          </Stacker.Stackable>
          <Stacker.Stackable className="deal-type-inputs__stackable">
            <Input
              id="Rebate"
              name="rebate"
              value={this.props.scenario.details.rebate}
              label="Rebate"
              onClick={() => this.openCloseModal("showRebatesModal")}
              onChange={this.onDetailsChange}
              onBlur={this.props.save}
              tabIndex={11}
            />
          </Stacker.Stackable>
          <Stacker.Stackable className="deal-type-inputs__stackable">
            <Input
              id="Fees"
              name="fees"
              value={this.props.scenario.details.fees}
              label="Fees"
              onClick={() => this.openCloseModal("showFeesModal")}
              onChange={this.onDetailsChange}
              onBlur={this.props.save}
              tabIndex={4}
            />
          </Stacker.Stackable>
          <Stacker.Stackable className="deal-type-inputs__stackable">
            <Input
              id="Allowance"
              name="allowance"
              value={this.props.scenario.trade.allowance}
              label="Trade Allowance"
              onChange={this.onTradeChange}
              onBlur={this.props.save}
              tabIndex={12}
            />
          </Stacker.Stackable>
          <Stacker.Stackable className="deal-type-inputs__stackable">
            <Input
              id="Taxes"
              name="taxes"
              value={this.props.scenario.details.taxes}
              label="Taxes"
              //eslint-disable-next-line
              onClick={() => this.openCloseModal("showTaxesModal")}
              onChange={this.onDetailsChange}
              onBlur={this.props.save}
              tabIndex={5}
            />
          </Stacker.Stackable>
          <Stacker.Stackable className="deal-type-inputs__stackable">
            <Input
              id="Payoff"
              name="payoff"
              value={this.props.scenario.trade.payoff}
              label="Trade Payoff"
              onChange={this.onTradeChange}
              onBlur={this.props.save}
              tabIndex={13}
            />
          </Stacker.Stackable>
          <Stacker.Stackable className="deal-type-inputs__stackable">
            <Input
              id="Rate"
              name="rate"
              value={this.props.scenario.details.rate}
              label="Rate %"
              onChange={this.onDetailsChange}
              onBlur={this.props.save}
              tabIndex={6}
            />
          </Stacker.Stackable>
          <Stacker.Stackable className="deal-type-inputs__stackable">
            <Input
              id="ACV"
              name="acv"
              value={this.props.scenario.trade.acv}
              label="Trade ACV"
              onChange={this.onTradeChange}
              onBlur={this.props.save}
              tabIndex={14}
            />
          </Stacker.Stackable>
        </Stacker>
        <Stacker.Stackable className="deal-type-inputs__stackable">
          <Input
            id="Term"
            name="term"
            value={this.props.scenario.details.term}
            label="Term"
            options={[
              { label: "Monthly" },
              { label: "Bi-Weekly" },
              { label: "Weekly" },
              { label: "Semi-Annually" },
              { label: "Annually" }
            ]}
            onChange={this.onDetailsChange}
            onBlur={this.props.save}
            tabIndex={7}
          />
        </Stacker.Stackable>
        <Stacker.Stackable className="deal-type-inputs__stackable">
          <Input
            id="LenderFee"
            name="LenderFee"
            value={this.props.scenario.trade.acv}
            label="Lender Fee"
            onChange={this.onTradeChange}
            onBlur={this.props.save}
            tabIndex={14}
          />
        </Stacker.Stackable>
        <Stacker.Stackable className="deal-type-inputs__stackable">
          <Input
            id="FirstPayment"
            name="firstPayment"
            value={this.props.scenario.details.firstPayment}
            label="Days to 1st Payment"
            onChange={this.onDetailsChange}
            onBlur={this.props.save}
            tabIndex={8}
          />
        </Stacker.Stackable>
        <Stacker.Stackable className="deal-type-inputs__stackable">
          <ReadOnly
            id="Subtotal"
            label="Subtotal"
            default="-"
            value={this.props.scenario.summary.subtotal}
          />
        </Stacker.Stackable>
        <Taxes
          name="taxes"
          //eslint-disable-next-line
          closeDialog={() => this.openCloseModal("showTaxesModal")}
          dialogStatus={this.state.showTaxesModal}
          populateValue={this.onDialogChange}
        />
        <Fees
          dealId={123}
          zipCode={123456}
          scenarioId={this.props.scenario.id}
          name="fees"
          //eslint-disable-next-line
          closeDialog={() => this.openCloseModal("showFeesModal")}
          dialogStatus={this.state.showFeesModal}
          populateValue={this.onDialogChange}
        />
        <Rebates
          name="rebate"
          //eslint-disable-next-line
          closeDialog={() => this.openCloseModal("showRebatesModal")}
          dialogStatus={this.state.showRebatesModal}
          populateValue={this.onDialogChange}
        />
      </div>
    );
  }
}

Balloon.propTypes = {
  scenario: PropTypes.instanceOf(Scenario).isRequired,
  actions: PropTypes.object.isRequired,
  save: PropTypes.func.isRequired
};

export default Balloon;
