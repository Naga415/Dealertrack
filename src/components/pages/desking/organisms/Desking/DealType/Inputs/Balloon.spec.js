'use strict';
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import Balloon from './Balloon';
import Scenario from '../../../../../../../models/Scenario';
import BalloonDetails from '../../../../../../../models/BalloonDetails';
import BalloonSummary from '../../../../../../../models/BalloonSummary';
import * as ScenarioTypes from '../../../../../../../models/ScenarioTypes';
import TradeDetails from '../../../../../../../models/TradeDetails';


const primaryScenario = () => {
  const tradeDetails = new TradeDetails();
  tradeDetails.allowance = 20000;
  tradeDetails.payoff = 30000;
  tradeDetails.acv = 40000;

  const balloonDetails = new BalloonDetails();
  balloonDetails.msrp = null;
  balloonDetails.sellingPrice = null;
  balloonDetails.addOns = null;
  balloonDetails.fees = null;
  balloonDetails.taxes = null;
  balloonDetails.lenderFee = null;
  balloonDetails.rebate = null;
  balloonDetails.rate = null;
  balloonDetails.term = null;
  balloonDetails.firstPayment = null;
  balloonDetails.downpayments = [];
  balloonDetails.Trade = tradeDetails;

  const balloonSummary = new BalloonSummary();
  balloonSummary.payment = 350;
  balloonSummary.amountFinanced = 38623;


  const scenario = new Scenario();
  scenario.type = ScenarioTypes.BALLOON;
  scenario.summary = balloonSummary;
  scenario.details = balloonDetails;
  return scenario;
};

const onDetailsChange = sinon.spy();
const onTradeChange = sinon.spy();
const onSave = sinon.spy();

function cleanUp() {
  onDetailsChange.reset();
  onTradeChange.reset();
  onSave.reset();
}

afterEach(() => {
  cleanUp();
});

describe('Balloon specs', () => {
  const props = {
    scenario: primaryScenario(),
    htmlId: "BalloonDeal",
    actions: {
      deal: {
        updateScenarioDetails: onDetailsChange,
        updateScenarioTrade: onTradeChange
      }
    },
    save: onSave
  };

  it('renders <Balloon />', () => {
    const component = renderer.create(<Balloon {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('triggers onChange, onSave events when details change', () => {
    const component = mount(<Balloon {...props} />);
    component.find('#MSRP').simulate('change');
    expect(onDetailsChange.calledOnce).toEqual(true);
    component.find('#MSRP').simulate('blur');
    expect(onSave.calledOnce).toEqual(true);
  });

  it('triggers onChange, onSave events when trade details change', () => {
    const component = mount(<Balloon {...props} />);
    component.find('#ACV').simulate('change');
    expect(onTradeChange.calledOnce).toEqual(true);
    component.find('#ACV').simulate('blur');
    expect(onSave.calledOnce).toEqual(true);
  });
});
