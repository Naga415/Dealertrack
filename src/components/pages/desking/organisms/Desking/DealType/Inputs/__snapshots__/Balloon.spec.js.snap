// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balloon specs renders <Balloon /> 1`] = `
<div
  className="deal-type-inputs"
  id={undefined}
>
  <div>
    <div
      className="deal-type-inputs__stackable"
    >
      <div
        className="desking-input"
      >
        <div
          className="desking-input__label"
        >
          <label
            className="desking-input__text-label control-label"
            htmlFor={undefined}
          >
            MSRP
          </label>
        </div>
        <input
          className="desking-input__value form-control input-sm"
          id="MSRP"
          inputMode="numeric"
          name="msrp"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="MSRP"
          readOnly={undefined}
          tabIndex={1}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="deal-type-inputs__stackable"
    >
      <div
        className="desking-input"
      >
        <div
          className="desking-input__label"
        >
          <div
            className="dropdown btn-group btn-group-sm btn-group-link"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="desking-input__dropdown dropdown-toggle btn btn-sm btn-link"
              disabled={false}
              id="Balloon"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              type="button"
            >
              Balloon
               
              <span
                className="caret"
              />
            </button>
            <ul
              aria-labelledby="Balloon"
              className="dropdown-menu"
              role="menu"
            >
              <li
                className=""
                role="presentation"
                style={undefined}
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  Percent
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  Dollar Amount
                </a>
              </li>
            </ul>
          </div>
        </div>
        <input
          className="desking-input__value form-control input-sm"
          id="Balloon"
          inputMode="numeric"
          name="Balloon"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Balloon"
          readOnly={undefined}
          tabIndex={9}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="deal-type-inputs__stackable"
    >
      <div
        className="desking-input"
      >
        <div
          className="desking-input__label"
        >
          <label
            className="desking-input__text-label control-label"
            htmlFor={undefined}
          >
            Selling Price
          </label>
        </div>
        <input
          className="desking-input__value form-control input-sm"
          id="SellingPrice"
          inputMode="numeric"
          name="sellingPrice"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Selling Price"
          readOnly={undefined}
          tabIndex={2}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="deal-type-inputs__stackable"
    >
      <div
        className="desking-input"
      >
        <div
          className="desking-input__label"
        >
          <div
            className="dropdown btn-group btn-group-sm btn-group-link"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="desking-input__dropdown dropdown-toggle btn btn-sm btn-link"
              disabled={false}
              id="Downpayment.0"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              type="button"
            >
              Down Payment
               
              <span
                className="caret"
              />
            </button>
            <ul
              aria-labelledby="Downpayment.0"
              className="dropdown-menu"
              role="menu"
            >
              <li
                className=""
                role="presentation"
                style={undefined}
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  Percent
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  Dollar Amount
                </a>
              </li>
            </ul>
          </div>
        </div>
        <input
          className="desking-input__value form-control input-sm"
          id="Downpayment.0"
          inputMode="numeric"
          name="downpayments.0"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Down Payment"
          readOnly={undefined}
          tabIndex={10}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="deal-type-inputs__stackable"
    >
      <div
        className="desking-input"
      >
        <div
          className="desking-input__label"
        >
          <button
            className="desking-input__link-label btn btn-link"
            disabled={false}
            type="button"
          >
            Add-Ons
          </button>
        </div>
        <input
          className="desking-input__value form-control input-sm"
          id="AddOns"
          inputMode="numeric"
          name="addOns"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Add-Ons"
          readOnly={undefined}
          tabIndex={3}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="deal-type-inputs__stackable"
    >
      <div
        className="desking-input"
      >
        <div
          className="desking-input__label"
        >
          <button
            className="desking-input__link-label btn btn-link"
            disabled={false}
            type="button"
          >
            Rebate
          </button>
        </div>
        <input
          className="desking-input__value form-control input-sm"
          id="Rebate"
          inputMode="numeric"
          name="rebate"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Rebate"
          readOnly={undefined}
          tabIndex={11}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="deal-type-inputs__stackable"
    >
      <div
        className="desking-input"
      >
        <div
          className="desking-input__label"
        >
          <button
            className="desking-input__link-label btn btn-link"
            disabled={false}
            type="button"
          >
            Fees
          </button>
        </div>
        <input
          className="desking-input__value form-control input-sm"
          id="Fees"
          inputMode="numeric"
          name="fees"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Fees"
          readOnly={undefined}
          tabIndex={4}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="deal-type-inputs__stackable"
    >
      <div
        className="desking-input"
      >
        <div
          className="desking-input__label"
        >
          <label
            className="desking-input__text-label control-label"
            htmlFor={undefined}
          >
            Trade Allowance
          </label>
        </div>
        <input
          className="desking-input__value form-control input-sm"
          id="Allowance"
          inputMode="numeric"
          name="allowance"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Trade Allowance"
          readOnly={undefined}
          tabIndex={12}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="deal-type-inputs__stackable"
    >
      <div
        className="desking-input"
      >
        <div
          className="desking-input__label"
        >
          <button
            className="desking-input__link-label btn btn-link"
            disabled={false}
            type="button"
          >
            Taxes
          </button>
        </div>
        <input
          className="desking-input__value form-control input-sm"
          id="Taxes"
          inputMode="numeric"
          name="taxes"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Taxes"
          readOnly={undefined}
          tabIndex={5}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="deal-type-inputs__stackable"
    >
      <div
        className="desking-input"
      >
        <div
          className="desking-input__label"
        >
          <label
            className="desking-input__text-label control-label"
            htmlFor={undefined}
          >
            Trade Payoff
          </label>
        </div>
        <input
          className="desking-input__value form-control input-sm"
          id="Payoff"
          inputMode="numeric"
          name="payoff"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Trade Payoff"
          readOnly={undefined}
          tabIndex={13}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="deal-type-inputs__stackable"
    >
      <div
        className="desking-input"
      >
        <div
          className="desking-input__label"
        >
          <label
            className="desking-input__text-label control-label"
            htmlFor={undefined}
          >
            Rate %
          </label>
        </div>
        <input
          className="desking-input__value form-control input-sm"
          id="Rate"
          inputMode="numeric"
          name="rate"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Rate %"
          readOnly={undefined}
          tabIndex={6}
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="deal-type-inputs__stackable"
    >
      <div
        className="desking-input"
      >
        <div
          className="desking-input__label"
        >
          <label
            className="desking-input__text-label control-label"
            htmlFor={undefined}
          >
            Trade ACV
          </label>
        </div>
        <input
          className="desking-input__value form-control input-sm"
          id="ACV"
          inputMode="numeric"
          name="acv"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Trade ACV"
          readOnly={undefined}
          tabIndex={14}
          type="number"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="deal-type-inputs__stackable"
  >
    <div
      className="desking-input"
    >
      <div
        className="desking-input__label"
      >
        <div
          className="dropdown btn-group btn-group-sm btn-group-link"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="desking-input__dropdown dropdown-toggle btn btn-sm btn-link"
            disabled={false}
            id="Term"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            type="button"
          >
            Term
             
            <span
              className="caret"
            />
          </button>
          <ul
            aria-labelledby="Term"
            className="dropdown-menu"
            role="menu"
          >
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                Monthly
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                Bi-Weekly
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                Weekly
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                Semi-Annually
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                Annually
              </a>
            </li>
          </ul>
        </div>
      </div>
      <input
        className="desking-input__value form-control input-sm"
        id="Term"
        inputMode="numeric"
        name="term"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Term"
        readOnly={undefined}
        tabIndex={7}
        type="number"
        value=""
      />
    </div>
  </div>
  <div
    className="deal-type-inputs__stackable"
  >
    <div
      className="desking-input"
    >
      <div
        className="desking-input__label"
      >
        <label
          className="desking-input__text-label control-label"
          htmlFor={undefined}
        >
          Lender Fee
        </label>
      </div>
      <input
        className="desking-input__value form-control input-sm"
        id="LenderFee"
        inputMode="numeric"
        name="LenderFee"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Lender Fee"
        readOnly={undefined}
        tabIndex={14}
        type="number"
        value=""
      />
    </div>
  </div>
  <div
    className="deal-type-inputs__stackable"
  >
    <div
      className="desking-input"
    >
      <div
        className="desking-input__label"
      >
        <label
          className="desking-input__text-label control-label"
          htmlFor={undefined}
        >
          Days to 1st Payment
        </label>
      </div>
      <input
        className="desking-input__value form-control input-sm"
        id="FirstPayment"
        inputMode="numeric"
        name="firstPayment"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Days to 1st Payment"
        readOnly={undefined}
        tabIndex={8}
        type="number"
        value=""
      />
    </div>
  </div>
  <div
    className="deal-type-inputs__stackable"
  >
    <div
      className="read-only"
    >
      <label
        className="read-only__label control-label"
        htmlFor={undefined}
      >
        Subtotal
      </label>
      <div
        className="read-only__value"
      >
        -
      </div>
    </div>
  </div>
</div>
`;
